discovery/common.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package discovery
+
+import (
+	"path/filepath"
+
+	"github.com/prometheus/common/config"
+)
+
+// SetHTTPClientConfigDirectory sets filepaths to be relative to dir.
+func SetHTTPClientConfigDirectory(c *config.HTTPClientConfig, dir string) {
+	SetTLSConfigDirectory(&c.TLSConfig, dir)
+	if c.BasicAuth != nil {
+		c.BasicAuth.PasswordFile = JoinDir(dir, c.BasicAuth.PasswordFile)
+	}
+	c.BearerTokenFile = JoinDir(dir, c.BearerTokenFile)
+}
+
+// SetTLSConfigDirectory sets filepaths to be relative to dir.
+func SetTLSConfigDirectory(c *config.TLSConfig, dir string) {
+	c.CAFile = JoinDir(dir, c.CAFile)
+	c.CertFile = JoinDir(dir, c.CertFile)
+	c.KeyFile = JoinDir(dir, c.KeyFile)
+}
+
+// JoinDir returns path relative to dir. If path is empty or absolute, it is return unchanged.
+func JoinDir(dir, path string) string {
+	if path == "" || filepath.IsAbs(path) {
+		return path
+	}
+	return filepath.Join(dir, path)
+}
```

Comment:
This file shouldn't exist. It was moved to the config package, but I guess a rebase left it behind... I guess I have one more rebase to do :\